introduce script to automate models and config preparation for ovms d… (#2771)
openvinotoolkit/model_server: demos/common/export_models/README.md

# Exporting GEN AI models for deployments in OpenVINO Model Server

> This script automated exporting models from Hugging Faces hub or fine-tunned in Pytorch format to the models repository for deployment for the model serving.

Reviewer: ```suggestion
This script automates exporting models from Hugging Face hub or fine-tunned in Pytorch format to the models repository for deployment of the model serving.
```

                                       [--max_num_batched_tokens MAX_NUM_BATCHED_TOKENS] [--max_num_seqs MAX_NUM_SEQS] [--cache_size CACHE_SIZE] [--target_device DEVICE]

options:
  -h, --help            show this help message and exit
  --model_repository_path MODEL_REPOSITORY_PATH
                        Where the model should be exported to
  --source_model SOURCE_MODEL
                        HF model name or path to the local folder with pytorch or OpenVINO model
  --model_name MODEL_NAME
                        Model name that should be used in the deployment. Equal to source_name if HF model name is used
  --weight-format PRECISION
                        precision of the exported model
  --config_file_path CONFIG_FILE_PATH
                        path to the config file
  --overwrite_models    version of the model
  --kv_cache_precision {u8}
                        u8 or empty (model default). Reduced kv cache precision to u8 lowers the cache size consumption.
  --block_size BLOCK_SIZE
                        32 for CPU and 16 for GPU
  --enable_prefix_caching

>                         this algorithm is used to cache the prompt tokens.

Reviewer: ```suggestion
                        This algorithm is used to cache the prompt tokens.
```

# Exporting GEN AI models for deployments in OpenVINO Model Server

> This script automates exporting models from Hugging Faces hub or fine-tunned in Pytorch format to the models repository for deployment for the model serving.

Reviewer 2: ```suggestion
This script automates exporting models from Hugging Faces hub or fine-tuned in PyTorch format to the models repository for deployment for the model serving.
```

In one step it prepares a complete set of resources in the models repository for a supported GenAI use case.

```bash

> pip install -q -r requirements.txt

Reviewer 3: Our automatic tests would fail here - please add wget for test files or cloning of repository if this file is expected to be tested in automatic tests.

> pip install -q -r demos/common/export_models/requirements.txt

Reviewer 2: I see that this readme is not exposed to sphinx, therefore I assume this readme file will be read on github. If so, I think the reader try to run these commands from within the context of demos/common/export_models directory.

> pip install -q -r demos/common/export_models/requirements.txt

Reviewer 3: Our automatic tests would fail here - please add wget for test files or cloning of repository if this file is expected to be tested in automatic tests.

                                       [--overwrite_models] [--kv_cache_precision {u8}] [--block_size BLOCK_SIZE] [--enable_prefix_caching] [--disable_dynamic_split_fuse]
                                       [--max_num_batched_tokens MAX_NUM_BATCHED_TOKENS] [--max_num_seqs MAX_NUM_SEQS] [--cache_size CACHE_SIZE] [--target_device DEVICE]

options:
  -h, --help            show this help message and exit
  --model_repository_path MODEL_REPOSITORY_PATH
                        Where the model should be exported to
  --source_model SOURCE_MODEL
                        HF model name or path to the local folder with pytorch or OpenVINO model
  --model_name MODEL_NAME
                        Model name that should be used in the deployment. Equal to source_name if HF model name is used
  --weight-format PRECISION
                        precision of the exported model
  --config_file_path CONFIG_FILE_PATH
                        path to the config file
  --overwrite_models    version of the model
  --kv_cache_precision {u8}
                        u8 or empty (model default). Reduced kv cache precision to u8 lowers the cache size consumption.
  --block_size BLOCK_SIZE

>                         32 for CPU and 16 for GPU

Reviewer 2: ```suggestion
                        32 for CPU and 16 for GPU (is the default, but should be adjusted depending on deployment machine)
```
